### ISSUE TITLE
Signing Assemblies

### ISSUE THREAD

**Issue by Andrei15193** [akatakritos/PygmentSharp#25]
Hi, I'm currently developing [CodeMap](https://github.com/Andrei15193/CodeMap) and I'm using your library to offer syntax highlighting to code blocks written in C# (e.g.: in example documentation elements) when generating the documentation website. You can see this with this specific type: [AssemblyDocumentationAddition.cs](https://github.com/Andrei15193/CodeMap/blob/master/CodeMap/DeclarationNodes/AssemblyDocumentationAddition.cs) and the generated page here: [AssemblyDocumentationAddition.html](https://andrei15193.github.io/CodeMap/dev/CodeMap.DeclarationNodes.AssemblyDocumentationAddition.html).

For the build I use with GitHub Actions I am signing the assemblies to provide a strong name so that people can verify if the library came from me, multiple versions of the same library can be used, also to be able to register the assembly in the GAC. Not sure why would anyone be doing any of these, but I'm doing that for other libraries I've been developing. It's mostly for the versioning aspect and identity aspects of it. You can read more about it here: https://docs.microsoft.com/dotnet/standard/assembly/strong-named

The issue I've just encountered is that your library is not signed meaning it is not strongly named. Strongly named assemblies can only reference other strongly named assemblies.

Would you consider strongly naming your library?

The way I handled the `.snk` file for my builds is that I stored it as a GitHub secret in Base64 of the binary content of the file, the build converts the Base64 string to a byte array then writes it as a file and then uses it for signing the assemblies. This way the contents of the `.snk` file are not publicly available, but still part of the build.

**Comment by akatakritos** [comment]
I've only ever had bad experiences with strong naming. I'm not ruled it out but I need to do a bit more research first on what the impact would be. Is this a breaking change for current consumers on .NET framework?

Would something like this solve your problem though? https://github.com/dsplaisted/strongnamer -- I've seen it recommended a few times. Let's you sign it yourself during the build.

**Comment by Andrei15193** [comment]
Thanks for the quick response! The library I'm working on is not yet released, it will be soon, I'll have a look at the strongnamer you've linked, hopefully it can solve the problem 😃

**Comment by Andrei15193** [comment]
I got to configure StrongNamer, it does the trick 😃, I've disabled StrongNamer by default and it's only enabled in GitHub Actions (it was introducing some compilation errors from generating a signed assembly for my main project and types weren't being resolved in dependent projects). I got to publishing the NuGet packages and there's no problem with them.

Thank you for your help! I'm no longer blocked by this.
